app/models/notification.rb

```diff
@@ -5,43 +5,46 @@ class Notification < ApplicationRecord
   belongs_to :reminder_template, class_name: "Experimentation::ReminderTemplate", optional: true
   has_many :communications
 
-  # We have 'sms' in our appointment reminder message keys due to legacy reasons, even though
-  # they also sometimes point to whatsapp messages
-  APPOINTMENT_REMINDER_MSG_PREFIX = "sms.appointment_reminders"
+  APPOINTMENT_REMINDER_MSG_PREFIX = "communications.appointment_reminders"
 
   validates :status, presence: true
   validates :remind_on, presence: true
   validates :message, presence: true
+  validates :purpose, presence: true
 
   enum status: {
     pending: "pending",
     scheduled: "scheduled",
     sent: "sent",
     cancelled: "cancelled"
   }, _prefix: true
+  enum purpose: {
+    covid_medication_reminder: "covid_medication_reminder",
+    experimental_appointment_reminder: "experimental_appointment_reminder",
+    missed_visit_reminder: "missed_visit_reminder"
+  }, _suffix: true
 
   scope :due_today, -> { where(remind_on: Date.current, status: [:pending]) }
 
   def localized_message
-    case subject
-    when Appointment
+    case purpose
+    when "missed_visit_reminder", "experimental_appointment_reminder"
```

Comment:
All of these strings for the `purpose` can be symbols...I'll just comment on this one but the others in this PR can also get changed over.  Since these are basically enums a symbol better represents that.

Reply:
I do prefer symbols...

Reply:
Changing to symbols didn't work as expected. I think you can use symbols for enums like `purpose[:missed_visit_reminder]` but it seems the actual values are strings.